learning-another-language/destructuring.md

```diff
@@ -0,0 +1,128 @@
+# Destructuring Objects
+
+## Introduction
+
+In JavaScript we will work with Objects regularly and at times we will want to work directly with specific key-value pairs within a larger object.  
+
+We can for example do:
+
+```javascript
+const isUserAuthorized = function(user) {
+  const id = user.id;
+  const name = user.name;
+  const authorized = user.authorized;
+  
+  if (authorized) {
+    console.log(`${id}: ${name} is authorized`);
+  } else {
+    console.log(`${id}: ${name} is not authorized`);
+  }
+};
+```
+
+However it can be time consuming to individually take elements from one object, like `user.id` in the example above, and store them in local variables.  We could repeatably type `user.id` and `user.name` and `user.authorized`, but this is time consuming as well. JavaScript has a shorthand notation for this, known as *destructuring*.
```

Comment:
```suggestion
However, it can be time consuming to individually take elements from one object, like `user.id` in the example above, and store them in local variables.  We _could_ repeatedly type `user.id` and `user.name` and `user.authorized` when we need to access them instead, but this is time consuming as well. 

Thankfully, JavaScript has a shorthand notation for this known as **destructuring**.
```